### PR TITLE
F/preserve data

### PR DESCRIPTION
**Pull request by Gnwin**
## OOP school library: preserve data

In this project:

- classes for Person, Student and Teacher were added according to ruby style guidelines,
- the person class was decorated.
- associations were setup between newly added classes
- basic UI was added enhancing interactivity with the command line app
- code was refactored to SOLID OOP principles
- data was persistent using json files

All requirements for this project have been met according to this document:

https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md

### CODE REVIEW

File: helpers/app_helper.rb [hunk 1]
```diff
@@ -109,7 +110,7 @@ def find_id
     until answered
       puts 'Please Select a person using the selection number for the person (number to the left of the name)'
       @rentals.map(&:person).to_set.each_with_index do |person, idx|
-        puts "Selection #{idx} - #{person.name}"
+        puts "Selection #{idx} - #{person[:name]}"
```
Comment (danielufeli) [inline helpers/app_helper.rb]: -  [ ] Nice working on your helper methods, kindly check this `block` i get the error in the screenshot when I try to List all rentals for a given person id.

image

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @Gnwin , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Clean Code ✔️
- Descriptive Readme ✔️
- No Linter Errors ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: helpers/app_helper.rb (hunk 1 above)
Comment (Gnwin) [reply helpers/app_helper.rb]: fixed

**Review by enkog — APPROVED** [review]
Hi @Gnwin 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**- [OPTIONAL] Please handle the case when `one or more of the files are missing (the app should work and show no errors)`. This is a requirement for the project, and the prior reviewer may have missed it.**
- Example: `return [] unless File.exists?("people.json")`

image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gnwin** [comment]
Thanks @enkog.

**Pull request merged**